dnf/cli/aliases.py

```diff
@@ -0,0 +1,173 @@
+# aliases.py
+# Resolving aliases in CLI arguments.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from dnf.i18n import _
+
+import collections
+import dnf.cli
+from dnf.conf.config import PRIO_DEFAULT
+import dnf.exceptions
+import libdnf.conf
+import logging
+import os
+import os.path
+
+logger = logging.getLogger('dnf')
+
+ALIASES_DROPIN_DIR = '/etc/dnf/aliases.d/'
+ALIASES_CONF_PATH = '/etc/dnf/aliases.d/ALIASES.conf'
+ALIASES_USER_PATH = '/etc/dnf/aliases.d/USER.conf'
+
+
+class AliasesConfig(object):
+    def __init__(self, path):
+        self._path = path
+        self._parser = libdnf.conf.ConfigParser()
+        self._parser.read(self._path)
+
+    @property
+    def enabled(self):
+        option = libdnf.conf.OptionBool(True)
+        try:
+            option.set(PRIO_DEFAULT, self._parser.getData()["main"]["enabled"])
+        except IndexError:
+            pass
+        return option.getValue()
+
+    @property
+    def aliases(self):
+        result = collections.OrderedDict()
+        section = "aliases"
+        if not self._parser.hasSection(section):
+            return result
+        for key in self._parser.getData()[section]:
+            value = self._parser.getValue(section, key)
+            if not value:
+                continue
+            result[key] = value.split()
+        return result
+
+
+class Aliases(object):
+    def __init__(self):
+        self.aliases = collections.OrderedDict()
+        self.conf = None
+        self.enabled = True
+
+        if self._disabled_by_environ():
+            self.enabled = False
+            return
+
+        self._load_main()
+
+        if not self.enabled:
+            return
+
+        self._load_aliases()
+
+    def _disabled_by_environ(self):
+        return 'DNF_ALIASES_DISABLED' in os.environ
```

Comment:
what if user sets DNF_ALIASES_DISABLED=0 ?
I'd also consider renaming to DNF_DISABLE_ALIASES (maybe nit-picking too much)